nginx/tests/test_unit.py

```diff
@@ -460,3 +452,132 @@ def test_emit_non_generic_tags_when_disabled(instance):
+                ('slabs', ['slabs']),
+                ('stream/limit_conns', ['stream', 'limit_conns']),
+                ('stream/server_zones', ['stream', 'server_zones']),
+                ('stream/zone_sync', ['stream', 'zone_sync']),
+                ('stream/upstreams', ['stream', 'upstreams']),
+            ],
+        ),
+        (
+            7,
+            False,
+            [
+                ('http/requests', ['requests']),
+                ('processes', ['processes']),
+                ('http/server_zones', ['server_zones']),
+                ('http/upstreams', ['upstreams']),
+                ('http/limit_conns', ['limit_conns']),
+                ('http/location_zones', ['location_zones']),
+                ('http/limit_reqs', ['limit_reqs']),
+                ('nginx', []),
+                ('ssl', ['ssl']),
+                ('http/caches', ['caches']),
+                ('connections', ['connections']),
+                ('resolvers', ['resolvers']),
+                ('slabs', ['slabs']),
+            ],
+        ),
+        (
+            2,
+            False,
+            [
+                ('http/requests', ['requests']),
+                ('processes', ['processes']),
+                ('http/server_zones', ['server_zones']),
+                ('http/upstreams', ['upstreams']),
+                ('nginx', []),
+                ('ssl', ['ssl']),
+                ('http/caches', ['caches']),
+                ('connections', ['connections']),
+                ('slabs', ['slabs']),
+            ],
+        ),
+    ],
+)
+def test_get_enabled_endpoints(check, instance_plus_v7, version, use_stream, expected_endpoints, caplog):
+    caplog.clear()
+    caplog.set_level(logging.DEBUG)
+    instance = deepcopy(instance_plus_v7)
+    instance['use_plus_api_stream'] = use_stream
+    instance['plus_api_version'] = version
+    check = check(instance)
+    check._perform_request = mock.MagicMock(side_effect=mocked_perform_request)
+    assert sorted(list(check._get_enabled_endpoints())) == sorted(expected_endpoints)
+
+    # Assert this log line is not emmitted because if it does then the method fell back to all endpoints
+    assert "Could not determine available endpoints from the API" not in caplog.text
+
+    LOG_LINES_TO_ASSERT = [
+        "Querying base API url",
+        "Querying http API url",
+        "Available endpoints are",
+        "Supported endpoints are",
+    ]
+    STREAM_LOG_LINE = "Querying stream API url"
+    if use_stream:
+        LOG_LINES_TO_ASSERT.append(STREAM_LOG_LINE)
+    else:
+        assert STREAM_LOG_LINE not in caplog.text
+
+    for log_line in LOG_LINES_TO_ASSERT:
+        assert log_line in caplog.text
+
+
+@pytest.mark.parametrize("only_query_enabled_endpoints", [(True), (False)])
+def test_only_query_enabled_endpoints(check, dd_run_check, instance_plus_v7, only_query_enabled_endpoints, caplog):
+    caplog.clear()
+    caplog.set_level(logging.DEBUG)
```

Comment:
`caplog` isn't being used in this check. Should it be removed here?

Reply:
sure!